src/utp_stream.cpp

```diff
@@ -1673,7 +1673,14 @@ bool utp_socket_impl::send_pkt(int const flags)
 	}
 
 	if (!m_stalled)
+	{
 		++p->num_transmissions;
+		m_timeout = clock_type::now() + milliseconds(packet_timeout());
```

Comment:
libutp appears to only update the timeout on sending when theres nothing already in flight, and also only for `ST_DATA` and `ST_FIN` packets. Adding a check for `m_bytes_in_flight == 0 && (p->size > p->header_size || (flags & pkt_fin))` before resetting the timeout could probabaly be done here if preferred.

https://github.com/bittorrent/libutp/blob/2b364cbb0650bdab64a5de2abb4518f9f228ec44/utp_internal.cpp#L992-L999

Reply:
I like adding that condition here, to only update the timeout if there are no outstanding packets already.

Reply:
Do you think the timeout should only update on ST_DATA and ST_FIN packets as well? I'm not sure of the reasoning behind it yet, and would only be adding it as its what libutp does.